src/spectral/utils/spectral-validator.js

```diff
@@ -46,7 +48,7 @@ const parseResults = function(results, debug) {
 };
 
 // setup: registers the oas2/oas3 formats, and attempts to load the ruleset file
-const setup = async function(spectral) {
+const setup = async function(spectral, defaultMode = false) {
```

Comment:
Is there anywhere `setup` would be called without specifying this value? It looks like the one place we call it, we provide the value. So this default seems redundant. If it can be called in other places, that's a different story!

Reply:
Okay, I found it 🙂 This happens [on this line](https://github.com/IBM/openapi-validator/blob/master/src/lib/index.js#L18) but we have the default value there as a variable - so I think it would be better to pass it in that let it be potentially redefined. This should be addressed before approval